src/models/ServerStyleSheet.js

```diff
@@ -2,218 +2,104 @@
-}
+/* this error is used for makeStyleTag */
+const sheetClosedErr =
+  process.env.NODE_ENV !== 'production'
+    ? `
+Can't collect styles once you've consumed a ServerStyleSheet's styles!
+ServerStyleSheet is a one off instance for each server-side render cycle.
+- Are you trying to reuse it across renders?
+- Are you accidentally calling collectStyles twice?
+`.trim()
+    : ''
+
+const streamBrowserErr =
+  process.env.NODE_ENV !== 'production'
+    ? 'Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.'
+    : ''
 
 export default class ServerStyleSheet {
   closed: boolean
+  global: StyleSheet
   instance: StyleSheet
 
   constructor() {
-    this.instance = StyleSheet.clone(StyleSheet.instance)
-    this.instance.isStreaming = false
+    this.instance = StyleSheet.global.clone()
+    this.closed = false
   }
 
   collectStyles(children: any) {
     if (this.closed) {
-      throw new Error(
-        process.env.NODE_ENV !== 'production'
-          ? "Can't collect styles once you've called getStyleTags!"
-          : ''
-      )
+      throw new Error(sheetClosedErr)
     }
+
     return (
       <StyleSheetManager sheet={this.instance}>{children}</StyleSheetManager>
     )
   }
 
-  close() {
-    clones.splice(clones.indexOf(this.instance), 1)
-    this.closed = true
-  }
-
   getStyleTags(): string {
     if (!this.closed) {
-      this.close()
+      this.closed = true
     }
 
     return this.instance.toHTML()
   }
 
   getStyleElement() {
     if (!this.closed) {
-      this.close()
+      this.closed = true
     }
 
     return this.instance.toReactElements()
   }
 
   interleaveWithNodeStream(readableStream: stream.Readable) {
-    if (__SERVER__) {
-      const ourStream = new stream.Readable()
-
-      // $FlowFixMe
-      ourStream._read = () => {}
-
-      this.instance.isStreaming = true
-
-      readableStream.on('data', chunk => {
-        ourStream.push(
-          this.instance.tags.reduce((html, tag) => {
-            if (!tag.isSealed()) {
-              html += tag.toHTML() // eslint-disable-line no-param-reassign
-            }
-
-            return html
-          }, '') + chunk
-        )
-      })
-
-      readableStream.on('end', () => {
-        ourStream.push(null)
-        this.close()
-      })
-
-      readableStream.on('error', err => {
-        ourStream.emit('error', err)
-        this.close()
-      })
-
-      return ourStream
-    } else {
-      throw new Error(
-        process.env.NODE_ENV !== 'production'
-          ? 'streaming only works in Node.js, please do not try to call this method in the browser'
-          : ''
-      )
+    if (!__SERVER__ || IS_BROWSER) {
+      throw new Error(streamBrowserErr)
     }
-  }
 
-  static create() {
-    return new StyleSheet(isLocal => new ServerTag(isLocal))
+    /* the tag index keeps track of which tags have already been emitted */
+    const { instance } = this
+    let instanceTagIndex = 0
```

Comment:
this might cause problems with streaming - will verify tomorrow

Reply:
As far as I can tell since I always refer to the latest tag and am forcing the Stylesheet to create a new one after this routine, an index should suffice to keep track of which tags haven’t been seen before

But I might be wrong 😅